src/Sentry/Protocol/Trace.cs

```diff
@@ -7,7 +7,7 @@ namespace Sentry.Protocol;
 /// <summary>
 /// Trace context data.
 /// </summary>
-public class Trace : ITraceContext, ISentryJsonSerializable, ICloneable<Trace>, IUpdatable<Trace>
+public class Trace : ITraceContext, IHasExtra, ISentryJsonSerializable, ICloneable<Trace>, IUpdatable<Trace>
```

Comment:
@aritchie if I'm not way off base with [this comment](https://github.com/getsentry/sentry-dotnet/issues/3873#issuecomment-2635291329), I think `Trace` doesn't need to implement `IHasExtra`... we can just add some SetData and GetData methods to this (or, more idiomatic for .NET, just add a Data property).